src/routes.py

```diff
@@ -0,0 +1,282 @@
+     db.session.add(week)
+     db.session.commit()
+     return render_template('weekly.html', week=week, user=current_user, avgidealWt=averageidealWt)
+
+ else:
+     if(len(current_user.weekli)==0):
+         return redirect(url_for('user', a=1))
+     return render_template('weekly.html', count =count, user=current_user)
+
+
+
+def allowed_file(filename):
+    return '.' in filename and \
+           filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+@app.route('/profile',  methods=['GET', "POST"])
+@login_required
+def profile():
+    image= current_user.image_file
+    print(image)
+    if request.method == "POST":
+        firstnamehtml = request.form['firstname']
+        lastnamehtml = request.form['lastname']
+        emailhtml = request.form['email']
+        
+        
+        if 'file' not in request.files:
+            flash('No file part', category="error")
+        
+        image_file=request.files['file']
+        print(image_file)
+        
+        if  image_file and allowed_file( image_file.filename):
+            filename = secure_filename( image_file.filename)
+            image_file.save(os.path.join(app.config['UPLOAD_FOLDER'], filename))
+            image_file=filename
+        else:
+            image_file=image    
+            
+        
+        
+        allemail=[user.email for user in USER.query.all()]
+        firstname=current_user.firstname
+        lastname=current_user.lastname
+        email=current_user.email
+
+
+        if len(firstnamehtml) < 3:
+            flash('frist name must be grater than 3 character', category="error")
+        elif len(lastnamehtml) < 1:
+            flash("lastname must be grater than 1 characters", category="error")
+        elif len(emailhtml) < 6:
+            flash("Email should be grater than 6 characters", category='error')
+        else:
+         if(firstnamehtml !=current_user.firstname or lastnamehtml != current_user.lastname or emailhtml !=current_user.email or image_file !=image):
+            
+                if(emailhtml in allemail and emailhtml !=current_user.email ):
+                 flash("Email already used! please Use new email!", category='error')
+                else:
+                 user=USER.query.filter_by(id=current_user.id).one()
+                 user.firstname=firstnamehtml
+                 user.lastname=lastnamehtml
+                 user.email=emailhtml
+                 user.image_file=image_file
+                    
+                 db.session.add(user)
+                 db.session.commit()
+                 if(image!=image_file or firstnamehtml !=firstname or lastnamehtml != lastname or emailhtml !=email):
+                     flash('Update successful!', category='success')
+                
+            
+    image_file=url_for('static', filename='profile_pics/'+ current_user.image_file)
+    return render_template("account.html", image_file=image_file, user=current_user)
+ 
+
```

Comment:
One thing to keep in mind, and this might be a bit nitpicky given it is a practice project, but always look at coding best practices and try to follow the standards. It will be really helpful in the future. There are small things all around that could be improved. Overall great work. Just need to practice more.